### ISSUE TITLE
No longer relevant

### ISSUE THREAD

**Issue by MorneZaayman** [MicrosoftDocs/appcenter-docs#1006]
The Push section is no longer available on app center.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 94127961-00eb-9ba7-e7dc-9780424be2fb
* Version Independent ID: 3f5a2a50-d693-8af5-f32e-cc012e3b2aea
* Content: [Sending Push Notifications - Visual Studio App Center](https://docs.microsoft.com/en-us/appcenter/push/send-notification#feedback)
* Content Source: [docs/push/send-notification.md](https://github.com/MicrosoftDocs/appcenter-docs/blob/live/docs/push/send-notification.md)
* Service: **vs-appcenter**
* Sub-service: **app-center-push**
* GitHub Login: @jwargo
* Microsoft Alias: **jowargo**

**Comment by jwargo** [comment]
@PloxsyBop yes, that's true because we announced that we are retiring the service. Is there a reason you created this issue?

**Comment by MorneZaayman** [comment]
> @PloxsyBop yes, that's true because we announced that we are retiring the service. Is there a reason you created this issue?

The feature is still mentioned in App Center:
![image](https://user-images.githubusercontent.com/25976312/80537874-0f729c80-89a5-11ea-985a-a2b5358742e8.png)

The documentation mentions a migration guide, but this is at the bottom:
![image](https://user-images.githubusercontent.com/25976312/80537946-2ca76b00-89a5-11ea-8bf4-773ae190291e.png)

So it seems to me you have retired a service but still reference it in your products, and have not provided a reasonable alternative as of yet. Should you not allow new apps to still use App Center Push until you have made the SDKs available for the migration to Notification Hubs. Currently Notifications Hubs still needs a lot more manual setup compared to what App Center Push required. The Notification Hub instructions available are not very clear. This especially applies to something like Xamarin Forms compared to Xamarin.Android and Xamarin.iOS native.

**Comment by King-of-Spades** [comment]
@PloxsyBop where are you copying that text from? I'm not seeing the "Use Additional services" in the doc you filed this issue from. 

Adding @elamalani for thoughts on the migration footer:
https://docs.microsoft.com/en-us/appcenter/migration/push/#migration

**Comment by elamalani** [comment]
@PloxsyBop We will update the docs and share the migration guides from App Center Push to Notification Hubs soon.

**Comment by jwargo** [comment]
@PloxsyBop The service is still operational, the only change is that new apps can't enable the service. For that reason, we left the references to Push in the product. We will remove references to push once we retire the service (which will likely be by the end of the calendar year).

We have provided a reasonable alternative, Azure Notification Hubs is a robust, time-tested service that you can use in your apps today. As @elamalani said, we'll publish an update to the migration guide very soon. We delayed because we wanted to implement some stuff on the Notification Hubs side to make it easier to migrate, and we're making progress in that area.

**Comment by MorneZaayman** [comment]
The "use additional services" text comes from the page that appears when you create an app in App Center. I don't agree that Notification Hubs is a _reasonable alternative_ as it stands currently, but more just one that _technically_ can be used. It requires much more manual setup to get working compared with App Center Push. To offer a true migration path, Microsoft should create a drop-in replacement, something like a NuGet that we can reference and then instantiate with an API similar to how App Center Push currently works. Just abandoning a product and telling your customers to use something else, which is not well documented or as simple to implement as your old product does not inspire confidence to base an enterprise application on your infrastructure. Who knows when you change things again and then we have to figure out and rewrite all our code to work with your changes.

A big reason why enterprises choose to base their solutions on the "Microsoft Stack" is the stability of it over the years. If this is going to no longer be the case, it may be that developers and the enterprises they work for will have to start considering alternatives.

**Comment by elamalani** [comment]
@PloxsyBop Thanks for letting us know. We will update the overview page and remove references to Push. 
I will also forward your feedback to the Notification Hubs team. They would be interested to learn about your feedback and improve the service going forward.

**Comment by MorneZaayman** [comment]
Thanks. I'm sorry that sounded like a bit of a rant, but it becomes very frustrating when during the development of a project, the APIs you rely on suddenly get changed, with no real migration path other than to learn a new system which works quite differently to what existed before. This adds unnecessary cost to the project, and also adds extra maintenance costs to other projects currently using those APIs that have now changed. 

Just as a side note, we were looking at Google's Firebase offering in the beginning and a major factor in **not** going with Google, was that they have a reputation for always changing their APIs. We want our developers working on our product, not constantly keeping up with the "API flavour of the month".

This issue can probably be closed as it doesn't relate directly to the article it was created on, but more to the entire App Center Push feature. Thank you for taking the time to reply to my comments.

**Comment by jwargo** [comment]
@PloxsyBop I understand, retiring the service isn't easy on any of us.  We know ANH isn't as easy to use as ACP. Azure Notification Hubs (ANH) is actually much more robust than App Center Push (ACP), Push actually runs on top of it, so that's why we decided to pick one over the other. Notification Hubs handles a much higher load, is faster, and is more flexible. 

We'd been working on a plan to make ANH more like ACP, and that's why we held off on the announcement. We've finalized the plan and will likely announce final plans soon.
